common/math/geometry/src/spline/hermite_curve.cpp

```diff
@@ -316,7 +316,11 @@ std::pair<double, double> HermiteCurve::get2DMinMaxCurvatureValue() const
   /**
    * @brief 0.1 is a sampling resolution of the curvature
    */
-  for (double s = 0; s <= 1; s = s + 0.1) {
+  const int sample_count = 10;
+  const double resolution = 1.0 / sample_count;
```

Comment:
Use constexpr for constants.
Also, use the @note Doxygen command to record why the constant has the value it does.